web-client/public/js/grnstate.js

```diff
@@ -1,3 +1,46 @@
+let currentNetwork = null;
+
+const annotateLinks = network => {
+    // TODO This duplicates logic that is done on the server side for an .xlsx spreadsheet.
+    //      Think of a way to consolidate it. Having discovered this, it seems like this should
+    //      be done on the client side because it rearranges data redundantly, for ease of display.
+    network.positiveWeights = [];
+    network.negativeWeights = [];
+
+    network.links.forEach(link => {
+        if (network.sheetType === "unweighted" && !link.value) {
+            link.value = 1;
+        }
+
+        if (link.value > 0) {
+            link.type = "arrowhead";
+            // link.stroke = "MediumVioletRed";   // GRNsight v1 magenta edge color
+            link.stroke = "rgb(195, 61, 61)";     // Node coloring-consistent red edge color
+            network.positiveWeights.push(link.value);
+        } else {
+            link.type = "repressor";
+            // link.stroke = "DarkTurquoise";     // GRNsight v1 cyan edge color
+            link.stroke = "rgb(51, 124, 183)";    // Node coloring-consistent blue edge color
+            network.negativeWeights.push(link.value);
+        }
+    });
+};
```

Comment:
The _import_ routines included this as part of the loading sequence. I chose to place this function in _grnstate.js_ because upon studying it, I noted that everything being done here is on the network—i.e., this is a 100% model-centric computation. So it felt right to make it part of the model module.